### DISCUSSION TITLE
Issue with setTimeout in Jet Client

### DISCUSSION THREAD

**Discussion by Patas106** [JetClient/jet-client-support#121] [General]
Dear Jet Client team,

I am encountering an issue with the setTimeout function in Jet Client while running a JavaScript script. My code was previously working fine, but now it throws the following error:

`TypeError: execute on io.jetclient.js.binding.s.setTimeout failed due to: no applicable overload found (overloads: [Method[public final int io.jetclient.js.binding.s.setTimeout(java.lang.Runnable)], Method[public final int io.jetclient.js.binding.s.setTimeout(java.lang.Runnable,int)]], arguments: [com.oracle.truffle.js.runtime.builtins.JSFunctionObject$Unbound@6c80db95 (Unbound), 20000 (Integer)])`

Here is the relevant portion of my script:

`await new Promise(resolve => setTimeout(resolve, 20000));`

I suspect the issue might be related to how setTimeout is implemented in Jet Client, as the error suggests it expects a java.lang.Runnable object instead of a standard JavaScript function.

Could you clarify:

Is there a change in how setTimeout is implemented in recent versions of Jet Client?
Does Jet Client use a specific JavaScript engine (e.g., GraalVM) that could be affecting the behavior of setTimeout?
How should I modify my code to make it compatible with Jet Client's setTimeout?
Thank you for your time and assistance. I appreciate any guidance you can provide.

Best regards,
Patrik Vacek

**Comment by Patas106** [comment]
I forgot to mention that I am using version of Intellij idea 2024.3 and in previous version it's okay.

**Comment by AntonShuvaev** [comment]
Thanks for reporting! This is a regression caused by recent changes in version 2024.3.16, where I restricted access to Java classes in the GraalVM-JS configuration. I’ve already fixed it, and the new version will be available today or by Monday.

If you need an immediate solution, you can downgrade to version 2024.3.15 by downloading it from [JetBrains Marketplace](https://plugins.jetbrains.com/plugin/21173-jetclient--the-ultimate-rest-client/versions/stable/) and manually installing it in your IDE.

**Reply by Patas106** [reply]
Thank you for the update. I appreciate the quick fix and will keep an eye out for the new version.
